clients/stellar-relay-lib/src/connection/connector/message_reader.rs

```diff
@@ -74,36 +89,25 @@ pub(crate) async fn poll_messages_from_stellar(
 }
 
 /// Returns Xdr format of the `StellarMessage` sent from the Stellar Node
-async fn read_message_from_stellar(
-	r_stream: &mut tcp::OwnedReadHalf,
-	timeout_in_secs: u64,
-) -> Result<Xdr, Error> {
+fn read_message_from_stellar(mut stream: TcpStream) -> Result<Xdr, Error> {
 	// holds the number of bytes that were missing from the previous stellar message.
 	let mut lack_bytes_from_prev = 0;
 	let mut readbuf: Vec<u8> = vec![];
 
-	let mut buff_for_peeking = vec![0; 4];
+	let mut buff_for_reading = vec![0; 4];
 	loop {
+		let stream_clone = stream.try_clone()?;
```

Comment:
I don't know how much of a cost is to clone the stream every time in the reading loop, would an `Arc<Mutex<TcpConecction>` be useful to avoid cloning? 
Then something like `tcp_stream.lock().unwrap();` would give you the stream.

Also could be needed for the other comment about `spawn_blocking` (in case that solution was needed).